android/src/main/java/com/swmansion/rnscreens/ScreenStack.java

```diff
@@ -367,4 +408,81 @@ private void setupBackHandlerIfNeeded(ScreenStackFragment topScreen) {
       mFragmentManager.addOnBackStackChangedListener(mBackStackListener);
     }
   }
+
+  // below methods are taken from
+  // https://github.com/airbnb/native-navigation/blob/9cf50bf9b751b40778f473f3b19fcfe2c4d40599/lib/android/src/main/java/com/airbnb/android/react/navigation/ScreenCoordinatorLayout.java#L43
+  // and are used to swap the order of drawing views when navigating forward with the transitions
+  // that are making transitioning fragments appear one on another. See more info in the comment to
+  // the linked class.
+  @Override
+  public void removeView(final View view) {
+    if (isDetachingCurrentScreen) {
```

Comment:
i dont understand the logic on when this property should be reset back to `true` -- can this be better explained somwhere in the comments

Reply:
I will add a small comment here too, but the main one will be in the only place in code when we change it to `true`.